cloudsdk/src/main/java/io/particle/android/sdk/cloud/EventsDelegate.java

```diff
@@ -108,6 +108,28 @@ void unsubscribeFromEventWithID(long eventListenerID) throws ParticleCloudExcept
         }
     }
 
+    @WorkerThread
+    void unsubscribeFromEventWithHandler(SimpleParticleEventHandler handler) throws ParticleCloudException {
+        synchronized (eventReaders) {
+            for (int i = 0; i < eventReaders.size(); i++) {
```

Comment:
Not exactly sure what are you suggesting here. IndexOfValue would require EventReader which I don't have thus I am looping through a whole list of EventReaders.